core/server/kustomization.go

```diff
@@ -40,19 +41,47 @@ func (cs *coreServer) ListKustomizations(ctx context.Context, msg *pb.ListKustom
 		return nil, defaultClusterNotFound{}
 	}
 
-	for _, ns := range nsList {
-		nsResult, err := listKustomizationsInNamespace(ctx, clustersClient, ns.Name)
-		if err != nil {
-			cs.logger.Error(err, fmt.Sprintf("unable to list kustomizations in namespace: %s", ns.Name))
+	newNextPageToken := ""
 
-			continue
+	// TODO: Once the UI handles pagination we can remove this if block.
+	// It was left like that so the UI doesn't brake with backend pagination changes
+	if msg.Pagination == nil {
```

Comment:
I like this approach but not exactly as a way to replace the condition. The idea with this condition is to either use or not the new pagination logic, based on checking the existence of pagination parameters, this is for the purpose of avoiding breaking the current communication logic between the UI and API. The example you are giving is using the new pagination logic regardless of the pagination parameters. This will always return the first page when the request doesn't have any pagination parameters, which is probably not too bad at this point if the `defaultPageLimit` is large enough. However, I feel like we need to discuss more if this pagination approach is actually the one we want/need, so far it feels a bit vague. That's why I want to leave the current logic intact. If we decide in the future that this is not what we want we can just simply remove any pagination params from the UI and that will be it.